### PR TITLE
Support edwards25519 curve in Noise protocol name

### LINKED ISSUE #5: Allow `Ed25519` with Noise
Currently if you set:

```
    const CURVE_NAME: &'static str = "Ed25519";
```
In `cyphernet::EcPk` and try to use that for Noise, you get:
```
thread '' panicked at 'not implemented: curve Ed25519 is not supported by the Noise library', /noise-framework-0.2.0/src/state.rs:275:17
```

We should either remove that associated constant, or not panic, since we now are supporting ECDH on Ed.

### PR DESCRIPTION
**Pull request by dr-orlovsky**
Closes #5

### CODE REVIEW

File: noise/src/state.rs [hunk 1]
```diff
@@ -271,6 +271,7 @@ impl<E: Ecdh, D: Digest> HandshakeState<E, D> {
         let curve_name = match E::Pk::CURVE_NAME {
             "Curve25519" => "25519",
             "Secp356k1" => "secp256k1",
```
Comment (cloudhead) [inline noise/src/state.rs:273]: Typo here FYI

**Comment by cloudhead** [comment]
Great :+1:

File: noise/src/state.rs (hunk 1 above)
Comment (dr-orlovsky) [reply noise/src/state.rs:273]: Sorry, where? `secp256k1` with lowercase letter is a typo by Lightning Labs, not be: that's the name they are using for the Noise handshake in Lightning (the only implementation using Secp256k1 for Noise I am aware of).

Comment (dr-orlovsky) [reply noise/src/state.rs:273]: Can you pls clarify this point so I can merge or fix this

Comment (cloudhead) [reply noise/src/state.rs:273]: It says "Secp_3_56k1"

Comment (dr-orlovsky) [reply noise/src/state.rs:273]: shit, thank you for spotting that out!!!

**Pull request merged**